libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayHistoryMapper.kt

```diff
@@ -5,7 +5,17 @@ import com.google.gson.internal.LinkedTreeMap
 import com.mapbox.base.common.logger.Logger
 import com.mapbox.base.common.logger.model.Message
 
-typealias CustomEventMapper = (String, LinkedTreeMap<*, *>) -> ReplayEventBase?
+/**
+ * Additional mapper that can be used with [ReplayHistoryMapper].
+ */
+interface CustomEventMapper {
+
+    /**
+     * Override to map your own custom events from history files,
+     * into [ReplayEventBase] for the [ReplayHistoryPlayer]
+     */
+    fun map(eventType: String, properties: LinkedTreeMap<*, *>): ReplayEventBase?
+}
 
 /**
  * This class is responsible for creating [ReplayEvents] from history data.
```

Comment:
Don't forget to support our Java clients :trollface: Missing `@JvmOverloads` The same thing happens in `ReplayRouteMapper` and in the old `ReplayRouteLocationEngine`